core/src/test/scala/org/apache/spark/util/UtilsSuite.scala

```diff
@@ -1510,9 +1510,16 @@ class UtilsSuite extends SparkFunSuite with ResetSystemProperties with Logging {
     conf.set(SHUFFLE_SERVICE_ENABLED, true)
     conf.set(SparkLauncher.SPARK_MASTER, "yarn")
     conf.set("spark.yarn.maxAttempts", "1")
+    conf.set(SERIALIZER, "org.apache.spark.serializer.KryoSerializer")
     assert(Utils.isPushBasedShuffleEnabled(conf) === true)
     conf.set("spark.yarn.maxAttempts", "2")
     assert(Utils.isPushBasedShuffleEnabled(conf) === true)
```

Comment:
My fault. The correct key should be for spark.yarn.maxAppAttempts. This was added during the time when we can not enable push based shuffle if there are multiple attempts configured. But now that constraints has been removed. Please also help change the key to spark.yarn.maxAppAttempts.